api/common.go

```diff
@@ -27,12 +27,26 @@ const (
 	DefaultDockerfileName string = "Dockerfile"
 )
 
-// byPrivatePort is temporary type used to sort types.Port by PrivatePort
-type byPrivatePort []types.Port
+// portOrdering is a temporary type used to sort types.Port by its fields
+type portOrdering []types.Port
```

Comment:
this needs to be telling which order you want to sort :) *should*, I understand is not always possible